src/dom.ts

```diff
@@ -1,22 +1,40 @@
 import { Attributes, DOM, IFiber } from "./type"
 import { isStr, LANE } from "./reconcile"
 
+const hasOwnProperty = Object.prototype.hasOwnProperty;
+
+const defaultObj = {} as const;
+
+const jointIter = <P extends Attributes>(
+  aProps = defaultObj as P,
+  bProps = defaultObj as P,
+  callback: (name: string, a: any, b: any) => void
+) => {
+  for (const name in aProps) {
+    if (hasOwnProperty.call(aProps, name)) {
```

Comment:
Can this be `name in aProps`? Maybe can save bytes.

Reply:
Modern JS engines do [some optimizations](https://benediktmeurer.de/2017/09/07/restoring-for-in-peak-performance/) for the `for ... in` loops, followed by `hasOwnProperty` check, so this might be needed for getting maximum performance

Reply:
I tried Object.keys. It will be faster and have fewer bytes.

Reply:
Hah, I've also discovered that `Object.keys` and `forEach` performs almost the same as `for ... in` loop. But your version  has an issue: it will iterate twice by properties that are both in `aProps` and `bProps`. Correct version is this one:

```ts
const jointIter = <P extends Attributes>(
  aProps = defaultObj as P,
  bProps = defaultObj as P,
  callback: (name: string, a: any, b: any) => void
) => {
	Object.keys(aProps).forEach((k) => {
    callback(k, aProps[k], bProps[k]);
  });
  Object.keys(bProps).forEach((k) => {
    if (!hasOwnProperty.call(aProps, k)) {
      callback(k, undefined, bProps[k]);
    }
  });
}
```
I'd also rather prefer returning `undefined` to the callback as this is consistent with your previous implementation

Reply:
You're right. Thank you. I'll revise it again.
> I'd also rather prefer returning undefined to the callback as this is consistent with your previous implementation.
What does this mean? I like your implementation better than mine.

Reply:
> What does this mean? I like your implementation better than mine.
It's just for compatibility with your previous implementation.

https://github.com/yisar/fre/blob/afea19f5b3d7ca91025f2f1c2d812f27c70397ee/src/dom.ts#L11
There is a problem with this check - it won't iterate any values at all if it will be called as `jointIter(null, { a: 1 }, callback)` or  `jointIter({ a: 1 }, null, callback)`

And just realized that my implementation has the flaw with `null` argument too - it will throw with any `null` argument. The right one will be like this:
```ts
const jointIter = <P extends Attributes>(
  aProps: P,
  bProps: P,
  callback: (name: string, a: any, b: any) => void
) => {
  aProps = aProps || defaultObj;
  bProps = bProps || defaultObj;
  Object.keys(aProps).forEach((k) => {
    callback(k, aProps[k], bProps[k]);
  });
  Object.keys(bProps).forEach((k) => {
    if (!hasOwnProperty.call(aProps, k)) {
      callback(k, undefined, bProps[k]);
    }
  });
}
```

Reply:
done! Thank u!